### PR TITLE
Add how to for mainnet subnet deploy using CLI

### PR DESCRIPTION
**Pull request by felipemadero**
Adds documentation of CLI subnet deploys to mainnet, using ledger and multisig

### CODE REVIEW

File: docs/subnets/multisig-deploy.md [hunk 1]
```diff
@@ -0,0 +1,400 @@
+# Deploy a Subnet on Mainnet with Multisig Authorization
+
+Connect ledger 1, be sure it is unblocked and with the Avalanche Ledger App Running, as in
```
Comment (holisticode) [inline docs/subnets/multisig-deploy.md:3]: as in?

File: docs/subnets/multisig-deploy.md [hunk 2]
```diff
@@ -0,0 +1,400 @@
...
+
+Subnet operations can be done under a permissioned multisig scheme,
+that is, a set of control addresses can be specified on subnet creation,
+which will be the only ones that can sign a valid tx related to that
+subnet (eg: create blockchain, add subnet validator).
+
+Furthermore, multisig threshold specifies how many of those control addresses
+are required to sign any such tx.
+
+The simplest case of no multisig, is having 1 control
```
Comment (holisticode) [inline docs/subnets/multisig-deploy.md:16]: simplest case without multisig

File: docs/subnets/multisig-deploy.md [hunk 3]
```diff
@@ -0,0 +1,400 @@
...
+one can specify 5 specific control keys, and a threshold of 2,
+and in this case, for every tx, 2 out of the 5 control keys will be needed to sign the tx. 
+
+When creating a given tx, a selection of the specific control keys to use (of threshold len) should be made.
+Afterwards, the tx should be signed by each one of this pre selected control keys subset.
+
+Eg, for the former case, 2 out of 5 control keys, the 5 control keys are defined in subnet creation,
+together with the enabling threshold. After that, a subnet op such as adding a subnet validator, first
+will be created by selecting which 2 of this 5 control addresses will be used for signing, and afterwards,
+the signature process can start, which needs to be fullfilled for those pre selected addresses.
```
Comment (holisticode) [inline docs/subnets/multisig-deploy.md:34]: typo: fulfilled

File: docs/subnets/multisig-deploy.md [hunk 4]
```diff
@@ -0,0 +1,400 @@
...
+  one of the two required subnet authorization control keys
+
+Sign command related ones:
+
+- Sign the blockchain creation tx with other one of the two required subnet authorization control keys
+- Issue the blockchain creation tx
+
+### Notice
+
+1. All operations will use ledger, as is required for mainnet. Stored keys can be used for multisig on fuji
```
Comment (holisticode) [inline docs/subnets/multisig-deploy.md:55]: Locally stored keys. A key is also stored inside a ledger for example.

File: docs/subnets/multisig-deploy.md [hunk 5]
```diff
@@ -0,0 +1,400 @@
...
+- Control key 4: `P-avax1g4eryh40dtcsltmxn9zk925ny07gdq2xyjtf4g`
+
+## Create and sign txs with the deploy command
+
+### Ledger 0
+
+Connect ledger 0 device, unblock it, and run the Avalanche Ledger Application, as described in [How to use Ledger](https://support.avax.network/en/articles/6150237-how-to-use-a-ledger-nano-s-or-nano-x-with-avalanche) (up to and including point 4).
+Remind that Ledger 0 should have at least 2 AVAX to pay for the tx fees.
+
+When using CLI, it should recognice the ledger and use the associated address `P-avax1kdzq569g2c9urm9887cmldlsa3w3jhxe0knfy5`
```
Comment (holisticode) [inline docs/subnets/multisig-deploy.md:93]: recognize

File: docs/subnets/multisig-deploy.md [hunk 6]
```diff
@@ -0,0 +1,400 @@
...
+Enter P-Chain address (Example: P-...): P-avax12gcy0xl0al6gcjrt0395xqlcuq078ml93wl5h8
+✔ Add
+Enter P-Chain address (Example: P-...): P-avax1g7nkguzg8yju8cq3ndzc9lql2yg69s9ejqa2af
+✔ Add
+Enter P-Chain address (Example: P-...): P-avax1g4eryh40dtcsltmxn9zk925ny07gdq2xyjtf4g
+✔ Done
+Your Subnet's control keys: [P-avax1wryu62weky9qjlp40cpmnqf6ml2hytnagj5q28 P-avax1kdzq569g2c9urm9887cmldlsa3w3jhxe0knfy5 P-avax12gcy0xl0al6gcjrt0395xqlcuq078ml93wl5h8 P-avax1g7nkguzg8yju8cq3ndzc9lql2yg69s9ejqa2af P-avax1g4eryh40dtcsltmxn9zk925ny07gdq2xyjtf4g]
+```
+
+Next is to specify threshold. Select 2:
```
Comment (holisticode) [inline docs/subnets/multisig-deploy.md:183]: the threshold

File: docs/subnets/multisig-deploy.md [hunk 7]
```diff
@@ -0,0 +1,400 @@
...
+chain creation tx, and deploy will fail, with the expected message
+
+```
+Error: wallet does not contain subnet auth keys
+exit status 1
+```
+
+### Specify specific control keys to be used to sign the chain creation tx
+
+After indicating 2 control keys are needed to sign any subnet tx, CLI ask the user
```
Comment (holisticode) [inline docs/subnets/multisig-deploy.md:205]: CLI asks

File: docs/subnets/multisig-deploy.md [hunk 8]
```diff
@@ -0,0 +1,400 @@
...
+```
+? Choose a subnet auth key: 
+  ▸ P-avax1wryu62weky9qjlp40cpmnqf6ml2hytnagj5q28
+    P-avax1kdzq569g2c9urm9887cmldlsa3w3jhxe0knfy5
+    P-avax12gcy0xl0al6gcjrt0395xqlcuq078ml93wl5h8
+    P-avax1g7nkguzg8yju8cq3ndzc9lql2yg69s9ejqa2af
+    P-avax1g4eryh40dtcsltmxn9zk925ny07gdq2xyjtf4g
+```
+
+Here, user should select one by one the two ledger addresses: `P-avax1kdzq569g2c9urm9887cmldlsa3w3jhxe0knfy5` and
```
Comment (holisticode) [inline docs/subnets/multisig-deploy.md:218]: the user

File: docs/subnets/multisig-deploy.md [hunk 9]
```diff
@@ -0,0 +1,400 @@
...
+    P-avax1kdzq569g2c9urm9887cmldlsa3w3jhxe0knfy5
+    P-avax12gcy0xl0al6gcjrt0395xqlcuq078ml93wl5h8
+    P-avax1g7nkguzg8yju8cq3ndzc9lql2yg69s9ejqa2af
+    P-avax1g4eryh40dtcsltmxn9zk925ny07gdq2xyjtf4g
+```
+
+Here, user should select one by one the two ledger addresses: `P-avax1kdzq569g2c9urm9887cmldlsa3w3jhxe0knfy5` and 
+`P-avax1g7nkguzg8yju8cq3ndzc9lql2yg69s9ejqa2af`
+
+Note: if the currently connected ledger address is not included here, operation will next fail with:
```
Comment (holisticode) [inline docs/subnets/multisig-deploy.md:221]: the operation

File: docs/subnets/multisig-deploy.md [hunk 10]
```diff
@@ -0,0 +1,400 @@
...
+✔ 2
+✔ P-avax1g7nkguzg8yju8cq3ndzc9lql2yg69s9ejqa2af
+✔ P-avax1g4eryh40dtcsltmxn9zk925ny07gdq2xyjtf4g
+Your subnet auth keys for chain creation: [P-avax1g7nkguzg8yju8cq3ndzc9lql2yg69s9ejqa2af P-avax1g4eryh40dtcsltmxn9zk925ny07gdq2xyjtf4g]
+*** Please sign subnet creation hash on the ledger device *** 
+Error: insufficient funds: provided UTXOs need 100000000 more units of asset "rgNLkDPpANwqg3pHC4o9aGJmf2YU4GgTVUMRKAdnKodihkqgr"
+exit status 1
+```
+
+A sucessfull input/ouput for ledger 0 connected, and ledger addresses selected for chain creation, will be:
```
Comment (holisticode) [inline docs/subnets/multisig-deploy.md:249]: typo: successful
Typoi; output

File: docs/subnets/multisig-deploy.md [hunk 11]
```diff
@@ -0,0 +1,400 @@
...
+
+On ledger a `Sign Hash` window will be active. Navigate to the ledger `Accept` window by using ledger's
+right button, and then authorize the request by pressing both left and right buttons.
+
+```
+Subnet has been created with ID: 2qUKjvPx68Fgc1NMi8w4mtaBt5hStgBzPhsQrS1m7vSub2q9ew. Now creating blockchain...
+*** Please sign blockchain creation hash on the ledger device *** 
+```
+
+After successfull subnet creation, user will be asked to sing the blockchain creation tx.
```
Comment (holisticode) [inline docs/subnets/multisig-deploy.md:275]: successful
the user
sign

File: docs/subnets/multisig-deploy.md [hunk 12]
```diff
@@ -0,0 +1,400 @@
...
+
+On ledger a `Sign Hash` window will be active. Navigate to the ledger `Accept` window by using ledger's
```
Comment (holisticode) [inline docs/subnets/multisig-deploy.md:277]: the ledger's

File: docs/subnets/multisig-deploy.md [hunk 13]
```diff
@@ -0,0 +1,400 @@
...
+right button, and then authorize the request by pressing both left and right buttons.
+
+On sucess, subnet deploy details will be provided. As only 1 address signed the chain creation tx, a
```
Comment (holisticode) [inline docs/subnets/multisig-deploy.md:280]: success

File: docs/subnets/multisig-deploy.md [hunk 14]
```diff
@@ -0,0 +1,400 @@
...
+| VM ID              | rW1esjm6gy4BtGvxKMpHB2M28MJGFNsqHRY9AmnchdcgeB3ii  |
++--------------------+----------------------------------------------------+
+
+1 of 2 required Blockchain Creation signatures have been signed. Saving tx to disk to enable remaining signing.
+
+Path to export partially signed tx to:
+```
+
+Enter name of file to write to disk, in this example, `partiallySigned.txt`. Should not exist on disk.
+CLI will ask until this condition is meet.
```
Comment (holisticode) [inline docs/subnets/multisig-deploy.md:300]: is met.

File: docs/subnets/multisig-deploy.md [hunk 15]
```diff
@@ -0,0 +1,400 @@
...
+```
+
+CLI ask the user to use another ledger, ledger 1 in this case which contains the address `P-avax1g7nkguzg8yju8cq3ndzc9lql2yg69s9ejqa2af` to continue the signing process with `transaction sign` command.
+
+## Sign sign the chain creation tx with the transaction sign command
+
+### Ledger 1
+
+Connect ledger 1 device, unblock it, and run the Avalanche Ledger Application, as described in [How to use Ledger](https://support.avax.network/en/articles/6150237-how-to-use-a-ledger-nano-s-or-nano-x-with-avalanche) (up to and including point 4).
+Remind that Ledger 1 does not need to have funds, will used to provide subnet auth only, not to pay fees.
```
Comment (holisticode) [inline docs/subnets/multisig-deploy.md:321]: Remember or remind you
and will be used to provide

File: docs/subnets/multisig-deploy.md [hunk 16]
```diff
@@ -0,0 +1,400 @@
...
+
+When using CLI, it should recognice the ledger and use the associated address `P-avax1g7nkguzg8yju8cq3ndzc9lql2yg69s9ejqa2af`
```
Comment (holisticode) [inline docs/subnets/multisig-deploy.md:323]: recognize

File: docs/subnets/multisig-deploy.md [hunk 17]
```diff
@@ -0,0 +1,400 @@
...
+as the one to do the second signature of the tx.
+
+### Issue the command and sign the tx
+
+Transaction in partiallySigned.txt will be recognized as mainnet one, and so ledger will be user automatically
+for signing. In fuji case, the user will be prompted to choose signing mechanism.
```
Comment (holisticode) [inline docs/subnets/multisig-deploy.md:329]: In the fuji case
the signing mechanism

File: docs/subnets/create-a-fuji-subnet.md [hunk 18]
```diff
@@ -89,7 +89,7 @@ All commands which issue a transaction require a private key loaded into the too
 
 :::warning
 
-The private key created for this tutorial should only be used for testing operations on `Fuji` or other testnets. Do NOT use this key on `Mainnet`. The key will be stored on your file system. Whoever gets access to that key will have access to any funds secured by that private key. In a future release, `Avalanche-CLI` will integrate with Ledger and support `Mainnet` operations.
+The private key created for this tutorial should only be used for testing operations on `Fuji` or other testnets. Do NOT use this key on `Mainnet`. The key will be stored on your file system. Whoever gets access to that key will have access to any funds secured by that private key. To deploy to Mainnet, follow [this tutorial](./create-a-mainnet-subnet).
```
[the remark this answers is not in the document]
Comment (felipemadero) [reply docs/subnets/create-a-fuji-subnet.md:92]: will keep formatted Fuji and Mainnet as those were used previously on the tutorials

File: docs/subnets/create-a-mainnet-subnet.md [hunk 19]
```diff
@@ -0,0 +1,577 @@
...
+- [`Avalanche-CLI`](https://github.com/ava-labs/avalanche-cli) installed
+
+## Virtual Machine
+
+Avalanche is a network composed of multiple blockchains. Each blockchain is an instance of a [Virtual Machine (VM)](../subnets/README.md#virtual-machines), much like an object in an object-oriented language is an instance of a class.
+That is, the VM defines the behavior of the blockchain.
+
+[Subnet-EVM](https://github.com/ava-labs/subnet-evm) is the VM that defines the Subnet Contract Chains. Subnet-EVM is a simplified version of [Avalanche C-Chain](https://github.com/ava-labs/coreth).
+
+This chain implements the Ethereum Virtual Machine and supports Solidity smart contracts as well as most other Ethereum client functionality.
```
[the remark this answers is not in the document]
Comment (felipemadero) [reply docs/subnets/create-a-mainnet-subnet.md:27]: This and the previous were taken from subnet-evm README

**Comment by dasconnor** [comment]
I'd like to take a pass over this before we merge it.

File: docs/subnets/multisig-deploy.md [hunk 20]
```diff
@@ -0,0 +1,563 @@
...
+    P-avax1kdzq569g2c9urm9887cmldlsa3w3jhxe0knfy5
+    P-avax12gcy0xl0al6gcjrt0395xqlcuq078ml93wl5h8
+    P-avax1g7nkguzg8yju8cq3ndzc9lql2yg69s9ejqa2af
+    P-avax1g4eryh40dtcsltmxn9zk925ny07gdq2xyjtf4g
+````
+
+As before, the user will need to indicate to use both ledger addresses for auth, and will need to have ledger 0
+connected in order to pay for the fee (0.001 AVAX) and be the first subnet auth signer.
+
+```
```
Comment (cgcardona) [inline docs/subnets/multisig-deploy.md:438]: Add a type here such as `bash`

File: docs/subnets/multisig-deploy.md [hunk 21]
```diff
@@ -0,0 +1,563 @@
...
+✔ P-avax1g7nkguzg8yju8cq3ndzc9lql2yg69s9ejqa2af
+Your subnet auth keys for add validator tx creation: [P-avax1kdzq569g2c9urm9887cmldlsa3w3jhxe0knfy5 P-avax1g7nkguzg8yju8cq3ndzc9lql2yg69s9ejqa2af].
+```
+
+Lets skip the validation node/period prompt description, as it is already described [here](./create-a-mainnet-subnet#add-a-avalidator). 
+
+Only note that start time for validator is not selected to be in one minute, to give more time to all the signing process
+to be finished.
+
+```
```
Comment (cgcardona) [inline docs/subnets/multisig-deploy.md:450]: Add a type here such as `bash`

File: docs/subnets/multisig-deploy.md [hunk 22]
```diff
@@ -0,0 +1,563 @@
...
+Partial tx created
+
+1 of 2 required Add Validator signatures have been signed. Saving tx to disk to enable remaining signing.
+
+Path to export partially signed tx to:
+```
+
+As the case is multisig, tx is not fully signed, and a file is asked to write into, lets use partialAddValidatorTx.txt
+
+```
```
Comment (cgcardona) [inline docs/subnets/multisig-deploy.md:493]: Add a type here such as `bash`

File: docs/subnets/multisig-deploy.md [hunk 23]
```diff
@@ -0,0 +1,563 @@
...
+Process is pretty similar to signing of chain creation tx. So some details will be omitted now.
+
+First, be sure that ledger 1 is connected, unblocked, and with Avalanche Ledger App running.
+
+### Issue the command to sign the tx
+
+Transaction in partialAddValidatorTx.txt will be recognized as `Mainnet` one, and so ledger will be used automatically
+for signing.
+
+```
```
Comment (cgcardona) [inline docs/subnets/multisig-deploy.md:516]: Add a type here such as `bash`

File: docs/subnets/multisig-deploy.md [hunk 24]
```diff
@@ -0,0 +1,563 @@
...
+
+## Sign and commit the add validator tx with the transaction commands
+
+Process is pretty similar to signing of chain creation tx. So some details will be omitted now.
+
+First, be sure that ledger 1 is connected, unblocked, and with Avalanche Ledger App running.
+
+### Issue the command to sign the tx
+
+Transaction in partialAddValidatorTx.txt will be recognized as `Mainnet` one, and so ledger will be used automatically
```
Comment (cgcardona) [inline docs/subnets/multisig-deploy.md:513]: Wrap `partialAddValidatorTx.txt` in backticks.

File: docs/subnets/multisig-deploy.md [hunk 25]
```diff
@@ -0,0 +1,563 @@
...
+
+```
+avalanche transaction sign testsubnet --input-tx-filepath partialAddValidatorTx.txt
+*** Please provide extended public key on the ledger device ***
+```
+
+On the ledger a `Provide Extended Public Key` window will be active. Navigate to the ledger `Accept` window by using
+the ledger's right button, and then authorize the request by pressing both left and right buttons.
+
+```
```
Comment (cgcardona) [inline docs/subnets/multisig-deploy.md:524]: Add a type here such as `bash`

File: docs/subnets/multisig-deploy.md [hunk 26]
```diff
@@ -0,0 +1,563 @@
...
+
+Next, a new signing will be started for the create chain tx.
+
+On the ledger a `Sign Hash` window will be active. Navigate to the ledger `Accept` window by using
+the ledger's right button, and then authorize the request by pressing both left and right buttons.
+
+After that the tx will be recognized as having all required signatures.
+
+```
```
Comment (cgcardona) [inline docs/subnets/multisig-deploy.md:536]: Add a type here such as `bash`

File: docs/subnets/multisig-deploy.md [hunk 27]
```diff
@@ -0,0 +1,563 @@
...
+
+Overwritting partialAddValidatorTx.txt
+
+Tx is fully signed, and ready to be committed
+
+Commit command:
+  avalanche transaction commit testsubnet --input-tx-filepath partialAddValidatorTx.txt
+```
+
+Now, partialAddValidatorTx.txt constains a fully signed tx, ready to be deployed.
```
Comment (cgcardona) [inline docs/subnets/multisig-deploy.md:548]: Wrap `partialAddValidatorTx.txt` in backticks

File: docs/subnets/multisig-deploy.md [hunk 28]
```diff
@@ -0,0 +1,563 @@
...
+
+### Issue the command to commit the tx
+
+```
```
Comment (cgcardona) [inline docs/subnets/multisig-deploy.md:552]: Add a type here such as `bash`

File: docs/subnets/multisig-deploy.md [hunk 29]
```diff
@@ -0,0 +1,563 @@
...
+avalanche transaction commit testsubnet --input-tx-filepath partialAddValidatorTx.txt
+```
+
+tx will be recognized as `Mainnet` one, and automatically deployed.
+
+```
```
Comment (cgcardona) [inline docs/subnets/multisig-deploy.md:558]: Add a type here such as `bash`

File: docs/subnets/create-a-mainnet-subnet.md [hunk 30]
```diff
@@ -0,0 +1,586 @@
...
+
+```bash
+✔ Manual
+
+To setup your node, you must do two things:
+
+1. Add your VM binary to your node's plugin directory
+2. Update your node config to start validating the subnet
+
+To add the VM to your plugin directory, copy or scp from /tmp/tGBrMADESojmu5Et9CpbGCrmVf9fiAJtZM5ZJ3YVDj5JTu2qw
```
[the remark this answers is not in the document]
Comment (felipemadero) [reply docs/subnets/create-a-mainnet-subnet.md:513]: this is inside a big triple backticks code block with the output of a command

File: docs/subnets/create-a-mainnet-subnet.md [hunk 31]
```diff
@@ -0,0 +1,586 @@
...
+
+If you installed avalanchego manually, your plugin directory is likely
+avalanchego/build/plugins.
```
[the remark this answers is not in the document]
Comment (felipemadero) [reply docs/subnets/create-a-mainnet-subnet.md:516]: this is inside a big triple backticks code block with the output of a command

File: docs/subnets/create-a-mainnet-subnet.md [hunk 32]
```diff
@@ -0,0 +1,586 @@
...
+
+If you start your node from the command line WITHOUT a config file (e.g. via command
+line or systemd script), add the following flag to your node's startup command:
+
+--whitelisted-subnets=2b175hLJhGdj3CzgXENso9CmwMgejaCQXhMFzBsm8hXbH2MF7H
```
[the remark this answers is not in the document]
Comment (felipemadero) [reply docs/subnets/create-a-mainnet-subnet.md:521]: this is inside a big triple backticks code block with the output of a command

File: docs/subnets/create-a-mainnet-subnet.md [hunk 33]
```diff
@@ -0,0 +1,586 @@
...
+(if the node already has a whitelisted-subnets config, append the new value by
```
[the remark this answers is not in the document]
Comment (felipemadero) [reply docs/subnets/create-a-mainnet-subnet.md:522]: same

File: docs/subnets/create-a-mainnet-subnet.md [hunk 34]
```diff
@@ -0,0 +1,586 @@
...
+comma-separating it).
+
+For example:
+./build/avalanchego --network-id=Mainnet --whitelisted-subnets=2b175hLJhGdj3CzgXENso9CmwMgejaCQXhMFzBsm8hXbH2MF7H
+
+If you start the node via a JSON config file, add this to your config file:
+whitelisted-subnets: 2b175hLJhGdj3CzgXENso9CmwMgejaCQXhMFzBsm8hXbH2MF7H
```
[the remark this answers is not in the document]
Comment (felipemadero) [reply docs/subnets/create-a-mainnet-subnet.md:529]: same

File: docs/subnets/create-a-mainnet-subnet.md [hunk 35]
```diff
@@ -0,0 +1,586 @@
...
+
+TIP: Try this command with the --avalanchego-config flag pointing to your config file,
```
[the remark this answers is not in the document]
Comment (felipemadero) [reply docs/subnets/create-a-mainnet-subnet.md:531]: same

File: docs/subnets/create-a-mainnet-subnet.md [hunk 36]
```diff
@@ -0,0 +1,586 @@
...
+this tool will try to update the file automatically (make sure it can write to it).
+
+After you update your config, you will need to restart your node for the changes to
+take effect.
+```
```
[the remark this answers is not in the document]
Comment (felipemadero) [reply docs/subnets/create-a-mainnet-subnet.md:536]: this is the end of the code block, that starts at line that contains Manual

**Review by yulin-dong — CHANGES_REQUESTED** [review]
Please fix merge conflict.

File: docs/subnets/create-a-mainnet-subnet.md [hunk 37]
```diff
@@ -0,0 +1,662 @@
...
+issue the TXs to create the Subnet and add validators.
+
+To get the first `Mainnet` address of your ledger device, first make sure it is connected,
+unblocked, and running the Avalanche app. Then execute the `key list` command:
+
+```bash
+avalanche key list --ledger 0 --mainnet
+```
+
+The command first asks the user to authorize the ledger to give the CLI extended public key
```
Comment (yulin-dong) [inline docs/subnets/create-a-mainnet-subnet.md:102]: There are mixed ledger and Ledger in the text. Should we use Ledger all the time?

Comment (dasconnor) [reply docs/subnets/create-a-mainnet-subnet.md]: I just pushed an update that fixes all non-code differences.

**Review by yulin-dong — APPROVED** [review]

**Review by cgcardona — APPROVED** [review]

**Pull request merged**